### DISCUSSION TITLE
Colors and Copy/Paste

### DISCUSSION THREAD

**Discussion by madprops** [madprops/grasshopper#77] [Announcements]
Colors can now be customized. You can edit and add as many colors as you want. Colors consist of `name` and `value`. Name can be `red` (one word, lowercase), and value is a color you pick with a color picker. 

Tabs can be copy pasted as described elsewhere. Commands have been added. And the item menu presents them in the `Copy` section.` Also ctrl+c, ctrl+x, and ctrl+v do this. Copy tabs, then select another tab and paste. The tabs are moved to its "approximate" location, then selected so you can adjust.

**Comment by madprops** [comment]
Copy paste needs an order fix, upload soon.

**Comment by N3C2L** [comment]
This is fascinating, many thanks! I am very excited to be able to try out the additions.

Thanks for the color adding feature. And that I can apply any color I want is awesome + naming and filter them is so useful.

The copy and paste feature is working well. Every time I tried to paste tabs, they get pasted always under the selected tab, thanks! I didn't need to adjust them after pasting them, that's great!

Observation: The right click extra menus doesn't have tooltips appearing. It might be useful for new users what they can do when right clicking tabs and hovering over the cmds.

**Comment by madprops** [comment]
Changing colors to work with any name, not just a single word

**Comment by N3C2L** [comment]
I observed an issue that's related to the new coloring feature:
If an existing color already present in the color menu has its name changed, the color will be deleted from previously assigned menus (Favorite bar, Hover button menu). However, if only the RGB sliders are altered, the colors will not be removed from these menus.

**Comment by madprops** [comment]
Yes colors are pretty simple, just name and value, they don't have a separate id, so the name acts as an id. I added however a `Replace Color` command to help on changing previous names.

**Comment by N3C2L** [comment]
Thanks for the replace color cmd!
Maybe a hint at the bottom on this menu, that the colors must have been added to the color menu before. Because I did the mistake, that I wrote a new color in the "new color" prompt, thinking I can create a new color there. But that color was not added before to the color menu. The result when clicked "submit" was, that nothing happened.

I think there is another issue: 
how to reproduce: 
1. for example, put a filter blue cmd in the favorite bar
2. run the replace color cmd and replace the blue color
expected result: the filter blue cmd should have been replaced by the new color and should be visible in the favorite bar

**Reply by madprops** [reply]
2) No, it doesn't replace command data, so it works as expected

**Reply by N3C2L** [reply]
Ok, thanks. I didn't know that this cmd replaces the selected color on all tabs that have assigned this color, and not in the menus.

**Reply by madprops** [reply]
I'm going to try giving colors ids, so they can be changed more easily, i'll see if that works

**Comment by madprops** [comment]
Ok colors work better now as they now have an internal id. Which means name and rgb can be changed freely and still keep references. This means whatever colors you had you'll have to make again to use the new structure. In `v2203`.

**Reply by N3C2L** [reply]
That's awesome, thanks! I tried to change the name of a color in the color menu, and the new name appeared instantly in the menus, for example the favorite bar.
